edk2toollib/acpi/ivrs_parser.py

```diff
@@ -0,0 +1,643 @@
+##
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+# Python script that converts a raw IVRS table into a struct
+##
+
+import os
+import sys
+import struct
+import xml.etree.ElementTree as ET
+
+IVRSParserVersion = '1.00'
+
+
+class IVRS_TABLE(object):
+    # Header Lengths
+    IVRSHeaderLength = 48
```

Comment:
Generally, these should be calculated from structure strings. That way the reader can easily see what fields contribute to the length and/or modify as needed.

Reply:
The constants are removed